File: app/src/main/java/com/example/tracker/weight/YearWeek.java
package com.example.tracker.weight;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;

public class YearWeek {

    private static final String DATE_PATTERN = "dd.MM.";

    private final Calendar calendar;

    public YearWeek(Calendar calendar) {
        int dayOfWeek = (calendar.get(Calendar.DAY_OF_WEEK) + 5) % 7;
        calendar.add(Calendar.DATE, -1 * dayOfWeek);
        this.calendar = calendar;
    }

    public static YearWeek now() {
        Calendar calendar = Calendar.getInstance();
        calendar.setFirstDayOfWeek(Calendar.MONDAY);
        calendar.setMinimalDaysInFirstWeek(4);
        return new YearWeek(calendar);
    }

    public void plusWeek() {
        calendar.add(Calendar.WEEK_OF_YEAR, 1);
    }

    public void minusWeek() {
        calendar.add(Calendar.WEEK_OF_YEAR, -1);
    }

    public String getDayRange() {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(DATE_PATTERN, Locale.GERMANY);
        String firstDay = simpleDateFormat.format(calendar.getTime());
        calendar.add(Calendar.DATE, 6);
        String lastDay = simpleDateFormat.format(calendar.getTime());
        calendar.add(Calendar.DATE, -6);
        return firstDay + " - " + lastDay;
    }

    public int getWeek() {
        return calendar.get(Calendar.WEEK_OF_YEAR);
    }
}

File: app/src/main/java/com/example/tracker/weight/YearWeek.kt
package com.example.tracker.weight

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale

class YearWeek(calendar: Calendar) {

    private val calendar: Calendar

    val dayRange: String
        get() {
            val simpleDateFormat = SimpleDateFormat(DATE_PATTERN, Locale.GERMANY)
            val firstDay = simpleDateFormat.format(calendar.time)
            calendar.add(Calendar.DATE, 6)
            val lastDay = simpleDateFormat.format(calendar.time)
            calendar.add(Calendar.DATE, -6)
            return "$firstDay - $lastDay"
        }

    val week: Int
        get() = calendar[Calendar.WEEK_OF_YEAR]

    val year: Int
        get() = calendar[Calendar.YEAR]

    fun plusWeek() {
        calendar.add(Calendar.WEEK_OF_YEAR, 1)
    }

    fun minusWeek() {
        calendar.add(Calendar.WEEK_OF_YEAR, -1)
    }

    companion object {
        private const val DATE_PATTERN = "dd.MM."

        fun now(): YearWeek {
            val calendar = Calendar.getInstance()
            calendar.firstDayOfWeek = Calendar.MONDAY
            calendar.minimalDaysInFirstWeek = 4
            return YearWeek(calendar)
        }
    }

    init {
        val dayOfWeek = (calendar[Calendar.DAY_OF_WEEK] + 5) % 7
        calendar.add(Calendar.DATE, -1 * dayOfWeek)
        this.calendar = calendar
    }
}